### PR TITLE
Paginate matches

### PR DESCRIPTION
**Pull request by lmachens**
Adds the possibility to filter for `username` and `gameId` on `/api/matches`.
In addition, the MongoDB collection has a schema to verify, that only valid matches exists. 
Some indexes optimize the performance of our new queries.

### CODE REVIEW

File: src/lib/router.ts [hunk 1]
```diff
@@ -76,12 +77,49 @@ router.post('/matches/:matchId/highlights', async (request, response, next) => {
...
+type PaginatedMatches = {
+  info: {
+    total: number;
+    itemsPerPage: number;
+    page: number;
+  };
+  results: Match[];
+};
+
+function createMatchesQuery(requestQuery: ParsedQs): Filter<Match> {
```
Comment (webdevelukas) [inline src/lib/router.ts]: Maybe we could put this function into `matches.ts`?

File: src/lib/router.ts [hunk 2]
```diff
@@ -76,12 +77,49 @@ router.post('/matches/:matchId/highlights', async (request, response, next) => {
...
+}
+
+router.get('/matches', async (request, response, next) => {
   try {
-    const matchesCollection = await getMatchesCollection();
-    const matches = await matchesCollection.find({}).toArray();
+    const page = Number(request.query.page) || 1;
+    const itemsPerPage = Number(request.query.itemsPerPage) || 10;
+    const matchesCollection = getMatchesCollection();
+    const query: Filter<Match> = createMatchesQuery(request.query);
```
Comment (webdevelukas) [inline src/lib/router.ts]: The type declaration `Filter<Match>` isn't necessary, because we know that `getMatchesCollection()` returns only that type.

**Review by webdevelukas — COMMENTED** [review]
🥇

**Review by webdevelukas — APPROVED** [review]
👍

**Review by AlexisRoe — APPROVED** [review]

**Pull request merged**